controllers/webhookcerts/controller.go

```diff
@@ -206,25 +180,10 @@ func (r *ReconcileWebhook) reconcileCerts(ctx context.Context, log logr.Logger)
 		return nil, err
 	}
 
-	for _, key := range []string{"tls.crt", "tls.key"} {
-		f := filepath.Join(r.certsDir, key)
-
-		data, err := ioutil.ReadFile(f)
-		if err != nil && !os.IsNotExist(err) {
-			return nil, err
-		}
-
-		if os.IsNotExist(err) || !bytes.Equal(data, cs.Data[key]) {
-			if err := ioutil.WriteFile(f, cs.Data[key], 0666); err != nil {
-				return nil, err
-			}
-		}
-	}
-
-	return cs.Data["ca.crt"], nil
+	return append(cs.Data["ca.crt"], cs.Data["ca.crt.old"]...), nil
```

Comment:
Why does this function now return a concatenation of all certificates instead of just the current one?

Reply:
It does not return a concatenation of all certificates, but of the current and the last public root certificate. 

If we did not do this, only the current certificate would be valid for the kubernetes webhook. However, on our webhook the certificates are updated only every 6 hours, in this time span the webhook would not work. So Kubernetes must have both the new and the old certificate stored in the MutatingWebhookConfiguration. Since it expires after a certain time anyway, kubernetes will ignore the old certificate once it expires.